lib/shortcuts/stream-lambda.js

```diff
@@ -59,6 +78,28 @@ class StreamLambda extends Lambda {
         StartingPosition
       }
     };
+    if (FilterCriteria) {
+      if (Object.prototype.toString.call(FilterCriteria) !== '[object Object]'){
+        throw new Error('`FilterCriteria` must be a JSON-like object');
+      }
+      if (!(FilterCriteria.Filters && Array.isArray(FilterCriteria.Filters))){
+        throw new Error('`FilterCriteria` must contain property `Filter` of type array');
+      }
+      if (FilterCriteria.Filters.length > 5){
+        throw new Error('`FilterCriteria.Filter` cannot contain more than 5 items, you may request a quota increase with AWS support if required.');
```

Comment:
And if someone requests a quota increase, they still won't be able to use this. Not sure if this one makes sense.

Reply:
I figures a quota increase would warrant updating this line, do we want to let the AWS error bubble up here?

Reply:
removed the quota check, we'll let the AWS error surface here instead